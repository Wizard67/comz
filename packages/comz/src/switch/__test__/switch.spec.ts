import { mount } from '@vue/test-utils'
import Switch from '../switch.vue'

describe('Switch.vue', () => {
  describe('props', () => {
    it('modelValue', () => {
      const wrapper = mount(Switch, {
        props: {
          modelValue: true
        }
      })

      expect(wrapper.classes()).toContain('cswitch--checked')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('disabled', () => {
      const wrapper = mount(Switch, {
        props: {
          modelValue: '',
          disabled: true
        }
      })

      expect(wrapper.classes()).toContain('cswitch--disabled')
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('events', () => {
    it('update:modelValue', async () => {
      const wrapper = mount(Switch, {
        props: {
          modelValue: false
        }
      })

      await wrapper.trigger('click')

      expect(wrapper.emitted()).toHaveProperty('update:modelValue')
      expect(wrapper.emitted()['update:modelValue'][0]).toEqual([true])
    })
  })

  describe('slots', () => {
    it('prepend', () => {
      const content = 'prepend'

      const wrapper = mount(Switch, {
        props: {
          modelValue: true
        },
        slots: {
          prepend: content
        }
      })

      const element = wrapper.find('.cswitch__prepend')

      expect(element.exists()).toBe(true)
      expect(element.text()).toBe(content)
      expect(wrapper.element).toMatchSnapshot()
    })

    it('append', () => {
      const content = 'append'

      const wrapper = mount(Switch, {
        props: {
          modelValue: true
        },
        slots: {
          append: content
        }
      })

      const element = wrapper.find('.cswitch__append')

      expect(element.exists()).toBe(true)
      expect(element.text()).toBe(content)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
