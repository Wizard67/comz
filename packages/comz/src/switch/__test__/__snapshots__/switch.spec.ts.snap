// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch.vue props disabled 1`] = `
<div
  class="cswitch"
>
  <!--v-if-->
  <div
    class="cswitch__field cswitch__field--checked cswitch__field--disabled"
  />
  <!--v-if-->
</div>
`;

exports[`Switch.vue props modelValue 1`] = `
<div
  class="cswitch"
>
  <!--v-if-->
  <div
    class="cswitch__field cswitch__field--checked"
  />
  <!--v-if-->
</div>
`;

exports[`Switch.vue slots append 1`] = `
<div
  class="cswitch"
>
  <!--v-if-->
  <div
    class="cswitch__field cswitch__field--checked"
  />
  <div
    class="cswitch__append"
  >
    
    
    append
    
    
  </div>
</div>
`;

exports[`Switch.vue slots prepend 1`] = `
<div
  class="cswitch"
>
  <div
    class="cswitch__prepend"
  >
    
    
    prepend
    
    
  </div>
  <div
    class="cswitch__field cswitch__field--checked"
  />
  <!--v-if-->
</div>
`;
